docs/manual/mustcall-checker.tex

```diff
@@ -0,0 +1,164 @@
+  is a special polymorphic annotation that indicates a resource-aliasing relationship between
+  the annotated elements. Internally, the MustCall Checker treats it identically to \<@PolyMustCall>.
+
+\item[\refqualclasswithparams{checker/mustcall/qual}{CreatesObligation}{String value}]
+  is a declaration annotation that can be written on a method or constructor. It takes a single Java expression as an
+  argument, referred to as its ``target''. Intuitively, this annotation means that the annotated method
+  ``resets'' the must-call obligations of its target to the least upper bound of their current value and the default
+  must-call obligations for its declared type. This annotation is repeatable, if the copies have different targets.
+
+\item[\refqualclass{checker/mustcall/qual}{Owning}]
+  is a declaration annotation that can be written on a method parameter, method return type, or
+  field declaration. It indicates that the annotated element is the ``owning'' pointer to any resource,
+  and the MustCall Checker may never forget information about its obligations.
+
+\item[\refqualclass{checker/mustcall/qual}{NotOwning}]
+  is a declaration annotation that can be written on a method parameter, method return type, or
+  field declaration. It indicates that the annotated element is \emph{not} the ``owning'' pointer to any resource,
+  and the MustCall Checker may forget information about its obligations temporarily.
+
+\end{description}
+
+A subset of the infinite lattice of the type hierarchy is shown in Figure~\ref{fig-must-call-hierarchy}.
+
+\begin{figure}
+\includeimage{mustcall}{5cm}
+\caption{The subtyping relationship of (some of) the MustCall Checker's qualifiers.
+Qualifiers in gray are used internally by the type system but should
+never be written by a programmer.}
+\label{fig-must-call-hierarchy}
+\end{figure}
+
+Note that \<@MustCall(\{"a", "b"\}) Object> represents more possible values than
+\<@MustCall(\{"a"\}) Object> does, because \<@MustCall(\{"a"\}) Object> can only
+contain an object that needs to call \<a> or one that needs to call nothing --- an
+object that needs to call \<b> (or both \<a> and \<b>) is not permitted.
+\<@MustCall(\{"a", "b"\}) Object> represents all objects that need to
+call \<a>, \<b>, both, or neither; it cannot not represent an object that needs
+to call some other method.
+
+The default type in unannotated code is \<@MustCall(\{\})>.
+A completely unannotated program will always type-check without warnings.
+For the MustCall Checker to be useful, the programmer should supply at least one non-empty
+\<@MustCall> annotation, either in source code or in a stub file.
+
+Note that it is always safe to write \<@Owning> and \<@NotOwning> annotations anywhere in your
+code (they are only hints to the checker), though doing so may reduce precision.
+
+\sectionAndLabel{Example use}{must-call-example}
+
+If a user annotates a class declaration with a \<@MustCall(...)> annotation, any use of the class's
```

Comment:
This is a standard mechanism discussed at `upper-bound-for-use` in the manual.  Please state this and cross-reference that section, so it doesn't sound like this is special behavior of this checker.

Reply:
Do you think we should remove this specific documentation from this checker's manual section, given that it's covered elsewhere? Given that this is a common problem (you encountered when typechecking plume-util, for example, which is why this subsection exists at all), I would prefer to keep this example here, too, so that a user can find it regardless of where they look.

Reply:
If it's a common problem, please keep the text here too.